src/core/Microsoft.PowerToys.Settings.UI/ViewModels/GeneralViewModel.cs

```diff
@@ -32,6 +34,18 @@ public GeneralViewModel()
             try
             {
                 GeneralSettingsConfigs = SettingsUtils.GetSettings<GeneralSettings>(string.Empty);
+
+                if (Helper.CompareVersions(GeneralSettingsConfigs.PowertoysVersion, Helper.GetProductVersion()) < 0)
+                {
+                    // Update settings
+                    GeneralSettingsConfigs.PowertoysVersion = Helper.GetProductVersion();
+                    SettingsUtils.SaveSettings(GeneralSettingsConfigs.ToJsonString(), string.Empty);
+                }
+            }
+            catch (FormatException e)
+            {
+                // If there is an issue with the version number format, don't migrate settings.
+                Debug.WriteLine(e.Message);

```

Comment:
do we want to write stuff to the debug console In a live environment?

Reply:
Debug calls are not included on Release builds, however there should probably be a telemetry log here.